src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -205,7 +205,7 @@ private IOptionsSerializationService GetOptionsSerializationService(string langu
         => _lazyLanguageSerializationService.GetOrAdd(languageName, n => _workspaceServices.GetLanguageServices(n).GetRequiredService<IOptionsSerializationService>());
 
     public Checksum CreateParseOptionsChecksum(ParseOptions value)
-        => Checksum.Create(value, this);
+        => Checksum.Create((value, @this: this), static (tuple, writer) => tuple.@this.SerializeParseOptions(tuple.value, writer));
```

Comment:
no need for the specialized helper for ParseOptions all the way in teh chcksum layer.